### ISSUE TITLE
erlang 18.0 compatibility: type dict(), queue() undefined [JIRA: CLIENTS-524]

### ISSUE THREAD

**Issue by p5n** [basho/riak-erlang-client#220]
making this project as ejabberd dependency produces

ejabberd/trunk/src/ejabberd-15.06/deps/riakc/src/riakc_obj.erl:287: type dict() undefined

ejabberd/trunk/src/ejabberd-15.06/deps/riakc/src/riakc_pb_socket.erl:120: type queue() undefined

**Comment by danieldreier** [comment]
I'm also seeing problems with erlang 18; tests fail as run using `./rebar ct` but I don't know enough erlang to figure out why. I tried simply bumping the allowed version in rebar.config but that obviously wasn't enough.

**Comment by mremond** [comment]
When trying to build master, I get:

```
MacBook-Pro-de-Mickael:riak-erlang-client mremond$ make
./rebar get-deps
==> riak-erlang-client (get-deps)
Pulling riak_pb from {git,"git://github.com/basho/riak_pb",{tag,"2.1.0.2"}}
Cloning into 'riak_pb'...
==> riak_pb (get-deps)
Pulling protobuffs from {git,"git://github.com/basho/erlang_protobuffs.git",
                             {tag,"0.8.1p5"}}
Cloning into 'protobuffs'...
==> protobuffs (get-deps)
Pulling meck from {git,"git://github.com/basho/meck.git",{tag,"0.8.2"}}
Cloning into 'meck'...
==> meck (get-deps)
./rebar compile
==> meck (pre_compile)
==> meck (compile)
ERROR: OTP release 18 does not match required regex R15|R16|17
ERROR: compile failed while processing /Users/mremond/tmp/riak-erlang-client/deps/meck: rebar_abort
make: *** [compile] Error 1
```

**Comment by binarytemple** [comment]
+1

**Comment by binarytemple** [comment]
@danieldreier can you clarify the reason why you think the problem might be related to rebar not being rebar3 - as opposed to the rebar.conf / syntax not being compatible with Erlang 18 ?

**Comment by lukebakken** [comment]
Fixed by 89e630ad0fdb8a42d887be21fcaf969473ab3dd7

**Comment by azhai** [comment]
Marked
